tests/functional/object/test_object_browser_client_provider_ui.py

```
@@ -0,0 +1,512 @@
+                secret_name=client1_secret_name,
+            )
+            time.sleep(2)
+
+            # Verify S3 sign-in success
+            assert s3_login.is_signed_in(), "S3 login failed on client1"
+            logger.info("Successfully signed in to object browser on client1")
+
+            # Navigate to the bucket (click on bucket name)
+            logger.info("Navigating to bucket: %s", client1_bucket_name)
+            bucket_ui.do_click(
+                (f"//a[contains(text(), '{client1_bucket_name}')]", By.XPATH)
```

Ruff (0.15.15)

[error] Undefined name `By`

(F821)